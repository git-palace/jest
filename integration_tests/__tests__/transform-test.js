/**
 * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
 *
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree. An additional grant
 * of patent rights can be found in the PATENTS file in the same directory.
 */

'use strict';

jest.unmock('../runJest');

const {linkJestPackage, run} = require('../utils');
const path = require('path');
const runJest = require('../runJest');

const stripJestVersion = stdout =>
  stdout.replace(/Jest CLI v\d{1,2}\.\d{1,2}\.\d{1,2}/, '<<REPLACED>>');

describe('babel-jest', () => {
  const dir = path.resolve(__dirname, '..', 'transform/babel-jest');

  beforeEach(() => {
    run('npm install', dir);
    linkJestPackage('babel-jest', dir);
  });

  it('runs transpiled code', () => {
    // --no-cache because babel can cache stuff and result in false green
    const {json} = runJest.json(dir, ['--no-cache']);
    expect(json.success).toBe(true);
    expect(json.numTotalTests).toBeGreaterThanOrEqual(1);
  });

  it('instruments only specific files and collects coverage', () => {
    const {stdout} = runJest(dir, ['--coverage', '--no-cache']);
    expect(stdout).toMatch('covered.js');
    expect(stdout).not.toMatch('not-covered.js');
    expect(stdout).not.toMatch('excluded-from-coverage.js');
    // coverage result should not change
    expect(stripJestVersion(stdout)).toMatchSnapshot();
  });
});

describe('no babel-jest', () => {
  const dir = path.resolve(__dirname, '..', 'transform/no-babel-jest');

  it('fails with syntax error on flow types', () => {
    const {stderr} = runJest(dir, ['--no-cache']);
    expect(stderr).toMatch(/FAIL.*fails-with-syntax-error/);
    expect(stderr).toMatch('SyntaxError: Unexpected token :');
  });

  test('instrumentation with no babel-jest', () => {
    const {stdout} = runJest(dir, ['--no-cache', '--coverage']);
    expect(stdout).toMatch('covered.js');
    expect(stdout).not.toMatch('excluded-from-coverage.js');
    // coverage result should not change
    expect(stripJestVersion(stdout)).toMatchSnapshot();
  });
});
