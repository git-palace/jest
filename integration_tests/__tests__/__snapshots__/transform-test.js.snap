exports[`babel-jest instruments only specific files and collects coverage 1`] = `
"Using <<REPLACED>>, jasmine2, babel-jest
----------------------------|----------|----------|----------|----------|----------------|
File                        |  % Stmts | % Branch |  % Funcs |  % Lines |Uncovered Lines |
----------------------------|----------|----------|----------|----------|----------------|
 this-directory-is-covered/ |    83.33 |      100 |       50 |    83.33 |                |
  covered.js                |    83.33 |      100 |       50 |    83.33 |             18 |
----------------------------|----------|----------|----------|----------|----------------|
All files                   |    83.33 |      100 |       50 |    83.33 |                |
----------------------------|----------|----------|----------|----------|----------------|

"
`;

exports[`no babel-jest instrumentation with no babel-jest 1`] = `
"Using <<REPLACED>>, jasmine2
----------------------------|----------|----------|----------|----------|----------------|
File                        |  % Stmts | % Branch |  % Funcs |  % Lines |Uncovered Lines |
----------------------------|----------|----------|----------|----------|----------------|
 this-directory-is-covered/ |    83.33 |      100 |       50 |    83.33 |                |
  covered.js                |    83.33 |      100 |       50 |    83.33 |             18 |
----------------------------|----------|----------|----------|----------|----------------|
All files                   |    83.33 |      100 |       50 |    83.33 |                |
----------------------------|----------|----------|----------|----------|----------------|

"
`;
